R/database.R

```diff
@@ -2628,3 +2636,258 @@ get_gene_expression = function(metadata,
+                                  min_score = 40,
+                                  pass = TRUE,
+                                  with_chr_prefix = FALSE,
+                                  projection = "grch37"){
+  
+  #check remote configuration
+  remote_session = check_remote_configuration(auto_connect = TRUE)
+  
+  these_samples_metadata = dplyr::filter(these_samples_metadata, sample_id == this_sample_id)
+  if(!nrow(these_samples_metadata==1)){
+    stop("metadata does not seem to contain your this_sample_id or you didn't provide one")
+  }
+  
+  #get wildcards
+  tumour_sample_id = this_sample_id
+  unix_group = pull(these_samples_metadata, unix_group)
+  seq_type = pull(these_samples_metadata, seq_type)
+  genome_build = pull(these_samples_metadata, genome_build)
+  pairing_status = pull(these_samples_metadata, pairing_status)
+  
+  if(pairing_status == "matched"){
+    normal_sample_id = pull(these_samples_metadata, normal_sample_id)
+  }else{
+    normal_sample_id = config::get("unmatched_normal_ids")[[unix_group]][[seq_type]][[genome_build]]
+  }
+
+  #get samples from individual flat files
+  path_template = config::get("results_flatfiles")$sv_manta$template
+  
+  if(!remote_session){
+    path_template_full = paste0(config::get("project_base"), path_template)
+    bedpe_path = glue::glue(path_template_full)
+    if(!file.exists(bedpe_path)){
+      print(paste("missing: ", bedpe_path))
+      message("Cannot find file locally. If working remotely, perhaps you forgot to load your config (see below) or sync your files?")
+      message('Sys.setenv(R_CONFIG_ACTIVE = "remote")')
+    }
+  }else{
+    local_path_template = paste0(config::get("project_base", config = "remote"), path_template)
+    bedpe_path = glue::glue(local_path_template)
+      
+    #check if the requested file is on local machine, if not, get it!
+    if(!file.exists(bedpe_path)){
+      remote_path_template = paste0(config::get("project_base", config = "default"), path_template)
+      remote_bedpe_path = glue::glue(remote_path_template)
+      cat(paste0("Local file not found.\ntrying to copy requested file: ", remote_bedpe_path, "\n", "To: ", bedpe_path))
+      dirN = dirname(bedpe_path)
+      suppressMessages(suppressWarnings(dir.create(dirN, recursive = T)))
+      ssh::scp_download(ssh_session, remote_bedpe_path, dirN)
+    }
+  }
+  
+  #read sample flat-file
+  message(paste0("Reading ", this_sample_id, " from: ", bedpe_path))
+  bedpe_dat_raw = suppressMessages(read_tsv(bedpe_path, comment = "##"))
+    
+  #if returned manta calls data frame is empty, return and potentially move over to the next sample (if get_manta_sv_by_samples is called).
+  if(!nrow(bedpe_dat_raw==0)){
+    message(paste0("WARNING! No SV calls found in flat-file for: ", this_sample_id))
+    return()
+  }
+  
+  #if selected projection is different from the genome build (for a particular sample), add information that this sample needs to be lifted (by get_manta_by_samples).
```

Comment:
What happens if you run this on a sample from the icgc_dart unix group?

Reply:
We get SV calls returned for this sample;
```
> icgc_dart_sample = get_gambl_metadata() %>%
+   dplyr::filter(unix_group == "icgc_dart") %>%
+   sample_n(1) %>%
+   pull(sample_id)
                                                                                                                                                                                                                                                                 
> my_svs = get_manta_sv_by_sample(this_sample_id = icgc_dart_sample,
+                                 these_samples_metadata = get_gambl_metadata())
Reading SP116645 from: /projects/nhl_meta_analysis_scratch/gambl/results_local/icgc_dart/manta_current/99-outputs/bedpe/genome--hs37d5/somaticSV/SP116645--SP116644--matched.somaticSV.bedpe                                                                     
SP116645 flat-file is not avaialble in the selected projection, running liftover_bedpe...                                                                                                                                                                        
SP116645 successfully lifted to grch37
> head(my_svs, 10)
   CHROM_A   START_A     END_A CHROM_B   START_B     END_B NAME SOMATIC_SCORE STRAND_A STRAND_B TYPE FILTER VAF_tumour VAF_normal DP_tumour DP_normal tumour_sample_id normal_sample_id pair_status
1        1  12975734  12975737       1  12975837  12975837    .            45        +        -  DEL   PASS      0.250      0.000        52        95         SP116645         SP116644     matched
2       10  90731254  90731254      10  90731688  90731688    .            56        +        -  DEL   PASS      0.130      0.000        69        94         SP116645         SP116644     matched
3       13  72927411  72927418      13  72927489  72927489    .            62        +        -  DEL   PASS      0.395      0.056        38        89         SP116645         SP116644     matched
4       14  74278653  74278653      14  74451363  74451363    .            68        +        -  DUP   PASS      0.104      0.000       134       140         SP116645         SP116644     matched
5       14  84711735  84711737      14  84715223  84715225    .            73        +        -  DEL   PASS      0.205      0.000        88       111         SP116645         SP116644     matched
6       14 106667431 106667433      14 106681734 106681736    .            82        +        -  DEL   PASS      0.193      0.000        83       123         SP116645         SP116644     matched
7       14 106695732 106695732      14 106696312 106696312    .            59        -        -  BND   PASS      0.327      0.000        49        68         SP116645         SP116644     matched
8       14 106696782 106696784      14 106780314 106780316    .            75        +        -  DEL   PASS      0.265      0.000        34        77         SP116645         SP116644     matched
9       14 106781774 106781774      14 106783374 106783374    .            69        +        -  DEL   PASS      0.232      0.000        56        88         SP116645         SP116644     matched
10      14 106783723 106783723      14 106783902 106783902    .            61        +        -  DEL   PASS      0.246      0.000        69        88         SP116645         SP116644     matched
```

Reply:
So it runs liftOver from hs37d5 to grch37?

Reply:
Yes, it does.